Re-arrange slip-coin example (#117)
kubeflow/kfp-tekton: samples/flip-coin/README.md

is used to demonstrate the use of conditions.

* [Installing Tekton Pipelines on Kubernetes](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#installing-tekton-pipelines-on-kubernetes) 

* [Installing Tekton Pipelines on OpenShift](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#installing-tekton-pipelines-on-openshift)

There are two ways to generate `Tekton` YAML. One is using `dsl-compile-tekton` command line tool, another way is using `TektonCompiler` inside your pipeline script. Both will generate the same `Tekton` YAML file. Then apply the YAML to your cluster by using the `tkn` command.

* Using command line to generate YAML

    ```
    # Using kfp-tekton command line tool to generate `Tekton` YAML
    dsl-compile-tekton --py condition.py --output condition.yaml
    ```

* Using `TektonCompiler` to generate YAML
    ```
    # Replace the kfp compile code in the condition.py's `main`: 
    # From
    #   `kfp.compiler.Compiler().compile(flipcoin_pipeline, __file__ + '.yaml')`
    # To
    #   `from kfp_tekton.compiler import TektonCompiler`

>     #   `TektonCompiler().compile(flipcoin_pipeline, __file__.replace('.py', '.yaml'))`

Reviewer: I'm not sure do we really need these comments since we already replaced for them in our .py file. Having this might confuse the users.

Author: ok, sure, I will remove this.

# The flip-coin pipeline

The [flip-coin pipeline](https://github.com/kubeflow/pipelines/blob/master/samples/core/condition/condition.py)
is used to demonstrate the use of conditions.

* [Installing Tekton Pipelines on Kubernetes](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#installing-tekton-pipelines-on-kubernetes) 

* [Installing Tekton Pipelines on OpenShift](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#installing-tekton-pipelines-on-openshift)

> * [Configuring artifact storage](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#configuring-artifact-storage): When using S3 bucket, make sure to update your bucket name.

Reviewer: We don't need this anymore for the filp-coin example

Author: sure.

> There are two ways to generate `Tekton` YAML. One is using `DSL-COMPILE-TEKTON` command line tool, another way is using `TektonCompiler` inside your pipeline script. Both will generate the same `Tekton` YAML file. Then apply the YAML to your cluster by using the `tkn` command.

Reviewer: `dsl-compile-tekton` should be lower case

Author: done

* Using command line to generate YAML

    ```
    # Using kfp-tekton command line tool to generate `Tekton` YAML
    dsl-compile-tekton --py condition.py --output condition.yaml
    ```

* Using `TektonCompiler` to generate YAML
    ```
    # Replace the kfp compile code in the condition.py's `main`: 
    # From
    #   `kfp.compiler.Compiler().compile(flipcoin_pipeline, __file__ + '.yaml')`
    # To
    #   `from kfp_tekton.compiler import TektonCompiler`
    #   `TektonCompiler().compile(flipcoin_pipeline, __file__.replace('.py', '.yaml'))`
    python condition.py
    ```

```
# Install tasks, conditions and pipeline
kubectl apply -f condition.yaml

> # Check the pipeline is running in your cluster, should see somethig like `conditional-execution-pipeline` is running

Reviewer: ```suggestion
# Check the pipeline is running in your cluster, should see something like `conditional-execution-pipeline` is running
```

Author: ok

kubectl get pipeline

# Start the pipeline
tkn pipeline start conditional-execution-pipeline --showlog
```

> If it is success, you will see something like this:

Reviewer: ```suggestion
If it runs successfully, you should see something like this:
```

Author: done
